posthog/plugins/web.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+
+@dataclass
+class WebSource:
+    id: int
+    source: str
+    token: str
+    config_schema: List[dict]
+    config: dict
+
+
+def get_transpiled_web_source(id: int, token: str) -> Optional[WebSource]:
+    from posthog.models import PluginConfig, PluginSourceFile
+
+    response = (
+        PluginConfig.objects.filter(
+            id=id,
+            web_token=token,
+            enabled=True,
+            plugin__pluginsourcefile__filename="web.ts",
+            plugin__pluginsourcefile__status=PluginSourceFile.Status.TRANSPILED,
+        )
+        .values_list("id", "plugin__pluginsourcefile__transpiled", "web_token", "plugin__config_schema", "config")
+        .first()
+    )
+
+    if not response:
+        return None
+
+    return WebSource(*(list(response)))  # type: ignore
+
+
+def get_decide_web_js_inject(team) -> List[dict]:
+    """
+    :type team: posthog.models.Team
+    """
```

Comment:
Why not just use a type annotation?

Reply:
I'd love to, but this file is imported in both `models/plugin.py` and `api/decide.py`. Importing team directly puts us in a cyclotron. Importing it via `def get_decide_web_js_inject(team: "posthog.models.Team") -> List[dict]:` (or just `"Team"`) didn't work either: ` error: Name "Team" is not defined`.

This was the only way I found 🤷

Reply:
I see – grep for `if TYPE_CHECKING:`, that should work around this issue